lib/uklibparam/libparam.lds.S

```diff
@@ -1,12 +1,17 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
 #include <uk/libparam.h>
 
 SECTIONS {
 	. = ALIGN(8);
+	PROVIDE(UK_LIBPARAM_PARAMSECTION_STARTSYM = .);
 	.UK_LIBPARAM_PARAMSECTION_NAME : {
-		PROVIDE(UK_LIBPARAM_PARAMSECTION_STARTSYM = .);
-		KEEP(*(.UK_LIBPARAM_PARAMSECTION_NAME))
-		KEEP(*(.UK_LIBPARAM_PARAMSECTION_NAME.*))
-		PROVIDE(UK_LIBPARAM_PARAMSECTION_ENDSYM = .);
+		KEEP(*(SORT_BY_NAME(UK_LIBPARAM_PARAM_NAMEPREFIX*)));
 	}
+	PROVIDE(UK_LIBPARAM_PARAMSECTION_ENDSYM = .);
 }
-INSERT AFTER .rodata
+INSERT AFTER .data
```

Comment:
With comment https://github.com/unikraft/unikraft/pull/1471/commits/6b37f210a493c9275206404e1e780dc64809f1cb#diff-dba471d15bca2731fbc43ddfc22a50d7662c080713d8f2184325d3ae92bc5554R133 we probably can make the section read-only.